src/content/loaders/powerPizzaPodcastLoader.ts

```diff
@@ -0,0 +1,107 @@
+import { feedLoader } from '@ascorbic/feed-loader';
+import type { Loader, LoaderContext } from 'astro/loaders';
+import { DataEntry, z } from 'astro:content';
+import type { ZodObject, ZodSchema } from 'astro:schema';
+
+interface Link {
+    href: string;
+    description: string;
+}
+
+/**
+* Extract links from "Scontrino" content
+*
+* This parser decorates the default feed object, coming from the feed loader, with the list of links in the "scontrino".
+* It allows to reuse them across the project without errors.
+*/
+function parseLinks(episode: any): Link[] {
+    const links: Link[] = [];
+
+    const SCONTRINO_SEPARATOR = "+++ Scontrino +++";
+
+    let scontrino: string;
+    if (episode.data.description!.includes(SCONTRINO_SEPARATOR)) {
+        scontrino = episode.data.description!.split(SCONTRINO_SEPARATOR)[1];
+    } else {
+        scontrino = episode.data.description!;
+    }
+    const scontrinoLines = scontrino
+        .replaceAll("<br /><br />", "<br />")
+        .replaceAll("<br /></a>", "</a><br />")
+        .split("<br />")
+        .map((l) => l.trim());
+    const rawLinkIndexes = scontrinoLines
+        .map((l, i) => (l.startsWith("<a") ? i : -1))
+        .filter((i) => i !== -1);
+    for (const index of rawLinkIndexes) {
+        const rawLink = scontrinoLines[index];
+
+        // Supports multiple links with the same description
+        let descriptionIndex = index - 1;
+        while (
```

Comment:
this seems a bit confusing, I would rewrite this whole bit splitting it into more testable functions